File: frontend/src/services/adminService.js
import api from './api';

export const adminService = {
  getDashboard: async () => {
    const response = await api.get('/api/v1/admin/dashboard');
    return response.data;
  },
  getShipments: async (params = {}) => {
    const query = new URLSearchParams(params).toString();
    const response = await api.get(`/api/v1/admin/shipments?${query}`);
    return response.data;
  },
  getAgents: async (page = 0, size = 20) => {
    const response = await api.get(`/api/v1/admin/agents?page=${page}&size=${size}`);
    return response.data;
  },
  getUsers: async (page = 0, size = 20) => {
    const response = await api.get(`/api/v1/admin/users?page=${page}&size=${size}`);
    return response.data;
  },
  activateUser: async (id) => {
    const response = await api.patch(`/api/v1/admin/users/${id}/activate`);
    return response.data;
  },
  deactivateUser: async (id) => {
    const response = await api.patch(`/api/v1/admin/users/${id}/deactivate`);
    return response.data;
  },
  manualAssignAgent: async (trackingNumber, agentId) => {
    const response = await api.post(`/api/v1/admin/shipments/${trackingNumber}/assign/${agentId}`);
    return response.data;
  },
  overrideStatus: async (trackingNumber, status) => {
    const response = await api.post(`/api/v1/admin/shipments/${trackingNumber}/override-status?status=${status}`);
    return response.data;
  },
  // Zone actions
  createZone: async (zonePayload) => {
    const response = await api.post('/api/v1/admin/zones', zonePayload);
    return response.data;
  },
  updateZone: async (id, zonePayload) => {
    const response = await api.put(`/api/v1/admin/zones/${id}`, zonePayload);
    return response.data;
  },
  deleteZone: async (id) => {
    const response = await api.delete(`/api/v1/admin/zones/${id}`);
    return response.data;
  },
  // Area actions
  createArea: async (areaPayload, zoneId) => {
    const response = await api.post(`/api/v1/admin/areas?zoneId=${zoneId}`, areaPayload);
    return response.data;
  },
  updateArea: async (id, areaPayload, zoneId) => {
    const response = await api.put(`/api/v1/admin/areas/${id}?zoneId=${zoneId}`, areaPayload);
    return response.data;
  },
  deleteArea: async (id) => {
    const response = await api.delete(`/api/v1/admin/areas/${id}`);
    return response.data;
  },
  // Rate Card actions
  getRateCards: async (page = 0, size = 20) => {
    const response = await api.get(`/api/v1/admin/rate-cards?page=${page}&size=${size}`);
    return response.data;
  },
  createRateCard: async (rateCardPayload, pickupZoneId, deliveryZoneId) => {
    const response = await api.post(
      `/api/v1/admin/rate-cards?pickupZoneId=${pickupZoneId}&deliveryZoneId=${deliveryZoneId}`,
      rateCardPayload
    );
    return response.data;
  },
  updateRateCard: async (id, rateCardPayload, pickupZoneId, deliveryZoneId) => {
    const response = await api.put(
      `/api/v1/admin/rate-cards/${id}?pickupZoneId=${pickupZoneId}&deliveryZoneId=${deliveryZoneId}`,
      rateCardPayload
    );
    return response.data;
  },
  deleteRateCard: async (id) => {
    const response = await api.delete(`/api/v1/admin/rate-cards/${id}`);
    return response.data;
  },
  createShipmentForCustomer: async (customerId, payload) => {
    const response = await api.post(`/api/v1/admin/shipments?customerId=${customerId}`, payload);
    return response.data;
  },
};

File: deliverytracker/src/main/java/com/lastmile/deliverytracker/admin/service/AdminService.java
package com.lastmile.deliverytracker.admin.service;

import com.lastmile.deliverytracker.admin.dto.response.AgentAdminResponse;
import com.lastmile.deliverytracker.admin.dto.response.RateCardAdminResponse;
import com.lastmile.deliverytracker.admin.dto.response.ShipmentAdminResponse;
import com.lastmile.deliverytracker.admin.dto.response.UserAdminResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminService {

    /**
     * Returns a paginated view of all shipments with their admin summary
     * (trackingNumber, customerName, status, totalCharge, createdAt).
     */
    Page<ShipmentAdminResponse> getShipments(Pageable pageable);

    /**
     * Returns a paginated view of all delivery agents with availability
     * status and active assignment count.
     */
    Page<AgentAdminResponse> getAgents(Pageable pageable);

    /**
     * Returns a paginated view of all platform users (all roles).
     */
    Page<UserAdminResponse> getUsers(Pageable pageable);

    /**
     * Returns a paginated view of all rate cards with zone names.
     */
    Page<RateCardAdminResponse> getRateCards(Pageable pageable);

    /**
     * Activates a user account by setting isActive = true.
     *
     * @throws com.lastmile.deliverytracker.admin.exception.AdminUserNotFoundException
     *         if no user with the given id exists
     */
    void activateUser(Long userId);

    /**
     * Deactivates a user account by setting isActive = false.
     *
     * @throws com.lastmile.deliverytracker.admin.exception.AdminUserNotFoundException
     *         if no user with the given id exists
     */
    void deactivateUser(Long userId);

    void manualAssignAgent(String trackingNumber, Long agentId);

    void overrideStatus(String trackingNumber, com.lastmile.deliverytracker.shipment.enums.ShipmentStatus status);

    com.lastmile.deliverytracker.pricing.entity.Zone createZone(com.lastmile.deliverytracker.pricing.entity.Zone zone);
    com.lastmile.deliverytracker.pricing.entity.Zone updateZone(Long zoneId, com.lastmile.deliverytracker.pricing.entity.Zone zone);
    void deleteZone(Long zoneId);

    com.lastmile.deliverytracker.pricing.entity.Area createArea(com.lastmile.deliverytracker.pricing.entity.Area area, Long zoneId);
    com.lastmile.deliverytracker.pricing.entity.Area updateArea(Long areaId, com.lastmile.deliverytracker.pricing.entity.Area area, Long zoneId);
    void deleteArea(Long areaId);

    com.lastmile.deliverytracker.pricing.entity.RateCard createRateCard(com.lastmile.deliverytracker.pricing.entity.RateCard rateCard, Long pickupZoneId, Long deliveryZoneId);
    com.lastmile.deliverytracker.pricing.entity.RateCard updateRateCard(Long rateCardId, com.lastmile.deliverytracker.pricing.entity.RateCard rateCard, Long pickupZoneId, Long deliveryZoneId);
    void deleteRateCard(Long rateCardId);
}
